### PR TITLE
homework3 ver. 1.0

### PR DESCRIPTION
**Pull request by neme3is**

### CODE REVIEW

File: src/test/java/com/epam/tc/hw3/utils/TestUtils.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+            InputStream inputStream = TestUtils.class.getClassLoader().getResourceAsStream(propertiesFilename);
+            if (inputStream != null) {
+                properties.load(inputStream);
+            } else {
+                throw new FileNotFoundException(
+                    "property file '" + propertiesFilename + "' not found in the classpath");
+            }
+            username = properties.getProperty("user");
+            pass = properties.getProperty("password");
+            inputStream.close();
```
Comment (shoggy) [inline src/test/java/com/epam/tc/hw3/utils/TestUtils.java:24]: Do not do this manually, use try-with-resources instead

File: src/test/java/com/epam/tc/hw3/Hw3TestsBaseClass.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        "Metals & Colors", "Elements packs");
+    protected static final List<String> CHECKBOX_LOG_TEST_VALUES = List.of("Fire: condition changed to true",
+        "Wind: condition changed to false",
+        "Earth: condition changed to true", "Water: condition changed to false");
+    protected static final List<String> RADIOBUTTON_TEST_VALUES = List.of("metal: value changed to Selen",
+        "metal: value changed to Bronze",
+        "metal: value changed to Silver", "metal: value changed to Gold");
+    protected static final List<String> DROPDOWN_TEST_ELEMENTS = List.of("Colors: value changed to Yellow",
+        "Colors: value changed to Blue",
+        "Colors: value changed to Green", "Colors: value changed to Red");
```
Comment (shoggy) [inline src/test/java/com/epam/tc/hw3/Hw3TestsBaseClass.java:29]: Please move these data to data provider and pass it to test methods as arguments

File: src/test/java/com/epam/tc/hw3/exercise1/ExerciseOneTests.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+    @Test
+    public void ex1Test() {
+        SoftAssertions softAssertions = new SoftAssertions();
+
+        MainPageObject mainPageObject = new MainPageObject(webDriver);
+
+        mainPageObject.login(TestUtils.getUsername(), TestUtils.getPass());
+
+        softAssertions.assertThat(mainPageObject.getLoggedInUsername()).isEqualTo("ROMAN IOVLEV");
```
Comment (shoggy) [inline src/test/java/com/epam/tc/hw3/exercise1/ExerciseOneTests.java:22]: This string should be moved to the `.properties` file

File: src/test/java/com/epam/tc/hw3/exercise2/ExerciseTwoTests.java [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        // Select in dropdown Yellow
+        differentElementsPage.selectYellowColorInDropdownList();
+
+        //*Assert that:
+        //•for each checkbox there is an individual log row and value is corresponded to the status of checkbox*//*
+        List<WebElement> logElements = differentElementsPage.getLogRecords();
+        differentElementsPage.clickAllCheckboxButtons();
+        for (int i = 0; i < CHECKBOX_LOG_TEST_VALUES.size(); i++) {
+            softAssertions.assertThat(logElements.get(i).getText().contains(CHECKBOX_LOG_TEST_VALUES.get(i))).isTrue();
+        }
```
Comment (shoggy) [inline src/test/java/com/epam/tc/hw3/exercise2/ExerciseTwoTests.java:47]: Convert `List<WebElement>` to `List<String>` and compare using one assertion, not in the loop

**Review by shoggy — CHANGES_REQUESTED** [review]

File: src/test/java/com/epam/tc/hw3/utils/TestUtils.java (hunk 1 above)
Comment (neme3is) [reply src/test/java/com/epam/tc/hw3/utils/TestUtils.java:24]: fixed

File: src/test/java/com/epam/tc/hw3/exercise1/ExerciseOneTests.java (hunk 3 above)
Comment (neme3is) [reply src/test/java/com/epam/tc/hw3/exercise1/ExerciseOneTests.java]: done

File: src/test/java/com/epam/tc/hw3/Hw3TestsBaseClass.java (hunk 2 above)
Comment (neme3is) [reply src/test/java/com/epam/tc/hw3/Hw3TestsBaseClass.java]: done

File: src/test/java/com/epam/tc/hw3/exercise2/ExerciseTwoTests.java (hunk 4 above)
Comment (neme3is) [reply src/test/java/com/epam/tc/hw3/exercise2/ExerciseTwoTests.java]: done

**Review by shoggy — APPROVED** [review]

**Pull request merged**